scripts/build.py

```diff
@@ -0,0 +1,79 @@
+import os
+import platform
+import struct
+
+from cffi import FFI
+
+
+def abs_machine():
+    machine = platform.machine()
+
+    pointer_bits = struct.calcsize("P") * 8
+    if pointer_bits not in (32, 64):
+        raise Exception("Unsupported pointer size")
+
+    is_64 = pointer_bits == 64
+
+    # x86 based archs
+    if machine in ('AMD64', 'x86_64', 'i686-64', 'i386', 'i686', 'x86'):
+        return "x86_64" if is_64 else "i686"
+    # arm based archs
+    elif machine in ('aarch64', 'arm64', 'armv6l', 'armv7l'):
+        return "aarch64" if is_64 else "arm"
+    else:
+        raise Exception("Unsupported processor")
+
+
+ffibuilder = FFI()
+system = platform.system()
+machine = abs_machine()
+parent_dir = os.path.dirname(os.path.dirname(__file__))
+
+if system == "Windows":
+    if machine == "x86_64":
+        libdir = "./lib32"
+    elif machine == "i686":
+        libdir = "./lib64"
+    else:
+        libdir = "ERROR"
+elif system == "Darwin":
+    if machine == "x86_64":
+        libdir = "/Users/runner/work/_temp/install/lib"
+    else:
+        libdir = "ERROR"
+else:
+    if machine in ("x86_64", "arm", "aarch64"):
+        libdir = "/usr/local/lib"
+    else:
+        libdir = "ERROR"
+
+if libdir == "ERROR":
+    raise Exception("Unsupported platform")
+
+ffibuilder.set_source(
+    "curl_cffi._wrapper",
+    """
+        #include "shim.h"
+    """,
+    libraries=["curl-impersonate-chrome"] if system != "Windows" else ["libcurl"],
```

Comment:
I think we may miss some steps for building windows dll, the name, if correctly configured, should be libcurl-impersonate-chrome.dll

Reply:
Is this miss-configuration from `curl-impersonate` package?

Reply:
Yes, curl-impersonate, possibly missing `autoreconf` on Windows.

Reply:
Then, IIUC it's not blocker for merge this PR.